src/main/java/io/temporal/samples/hello/HelloPeriodic.java

```diff
@@ -106,31 +133,45 @@
             GreetingActivities.class,
             ActivityOptions.newBuilder().setStartToCloseTimeout(Duration.ofSeconds(10)).build());
 
-    // Create our workflow stub that can be used to continue this workflow as a new run
-    // We use this because we want each workflow execution to be a separate workflow invocation
-    private final GreetingWorkflow continueAsNew =
-        Workflow.newContinueAsNewStub(GreetingWorkflow.class);
+    private boolean exitRequested = false;
+
+    @Override
+    public void requestExit() {
+      exitRequested = true;
+    }
 
     @Override
-    public void greetPeriodically(String name) {
-
-      // Loop as many times as defined in CONTINUE_AS_NEW_FREQUENCY
-      /*
-       * Here we define how many times we want to execute our workflow activity periodically.
-       * The value is set to 10 for the sake of the example. In real applications it would make
-       * more sense to set this to a higher value that matches a business cycle (for example once
-       * per 24 hours, etc).
-       */
-      int CONTINUE_AS_NEW_FREQUENCY = 10;
-
-      for (int i = 0; i < CONTINUE_AS_NEW_FREQUENCY; i++) {
-        // execute our activity method and sleep for a random amount of time
-        int delayMillis = random.nextInt(10000);
-        activities.greet("Hello " + name + "! Sleeping for " + delayMillis + " milliseconds.");
-        Workflow.sleep(delayMillis);
+    public boolean greetPeriodically(String name) {
+
+      for (int i = 0; i < SingleWorkflowRunIterations; i++) {
+
+        // Compute the timing of the next iteration:
+        int delayMillis = (ScheduleConfig.PeriodTargetSecs * 1000)
+                          + random.nextInt(ScheduleConfig.PeriodVariationSecs * 1000)
+                          - (ScheduleConfig.PeriodVariationSecs * 500);
```

Comment:
I would rewrite this to use Duration class:

Duration.ofSeconds(PeriodTargetSecs).plus(Duration.ofSeconds(random.nextInt(ScheduleConfig.PeriodVariationSecs) ....

Reply:
I found this (i.e. the conversion to `Duration` at the end) shorter / easier to read.
I left it as is for now, but if you find it adds value, please let me know and I can make the change.